electron/src/keytarStorageListener.ts

```diff
@@ -6,23 +6,35 @@ import {
     setPassword,
 } from 'keytar';
 
+import { BiometricMain } from 'jslib-common/abstractions/biometric.main';
+
+const AuthRequiredSuffix = 'biometric';
+const AuthenticatedActions = ['getPassword'];
+
 export class KeytarStorageListener {
-    constructor(private serviceName: string) { }
+    constructor(private serviceName: string, private biometricService: BiometricMain) { }
 
     init() {
         ipcMain.on('keytar', async (event: any, message: any) => {
             try {
-                let val: string = null;
-                if (message.action && message.key) {
+                const serviceName = this.serviceName + '_' + message.keySuffix;
```

Comment:
ah thank you, I hadn't tested this change with upgrade scenario. We need to make sure `Bitwarden` still works.

Reply:
Actually, Never lock clients will still need to unlock once in order to upgrade those keys. I'll need to think through that a bit if we want to keep them never locked.

Reply:
I think a single unlock requirement for never-lock clients should be okay in order to accomplish this. I wouldn't want to overcomplicate an upgrade purely to avoid a one-time unlock requirement.

Reply:
The existing upgrade method only needed to be called in one additional place. That method is quite light -- it executes one read from secure storage either returns if an upgrade has already been done or upgrades.

We should be able to remove the upgrade logic next release. The impact is only the one unlock as you say.